src/fetch.rs

```diff
@@ -266,9 +266,21 @@ fn fuzzy_query_registry_index(
     registry_path: impl AsRef<Path>,
 ) -> Result<Vec<CrateVersion>> {
     let crate_name = crate_name.into();
+    let remotes = PathBuf::from("refs/remotes/origin/");
+    let checkout_dir = registry_path.as_ref().join(".git").join(&remotes);
+    let checkout = checkout_dir
+        .read_dir()?
+        .next() // Is there always only one branch? (expecting either master og HEAD)
+        .ok_or_else(|| ErrorKind::MissingRegistraryCheckout(checkout_dir))??
+        .path();
     let repo = git2::Repository::open(registry_path)?;
     let tree = repo
-        .find_reference("refs/remotes/origin/master")?
+        .find_reference(
+            remotes
+                .join(checkout.file_name().unwrap()) //unwrap here is ok as we have know there is a file there
```

Comment:
why do we convert checkout to path and then to file_name? there is https://doc.rust-lang.org/std/fs/struct.DirEntry.html#method.file_name on `DirEntry`

Reply:
I do not know either. Thanks for that.